src/templates/_special/csrf-input.twig

```diff
@@ -0,0 +1,21 @@
+{{ hiddenInput(input.name, input.value, input.options) }}
```

Comment:
Why include this at all? And why check for `async` if the template will only ever get loaded when `$async` is `true`?

Reply:
> Why include this at all?

Just less work for JS to do. Any hypothetically if the fetch choked for some reason, you'd still end up with a CSRF that _might_ be valid.

> And why check for async if the template will only ever get loaded when $async is true?

It always uses the template.

Reply:
> Just less work for JS to do. Any hypothetically if the fetch choked for some reason, you'd still end up with a CSRF that _might_ be valid.

Eh, that feels gross to me. Seems like it would be better to fail hard, then to leak someone else’s CSRF token in the cached responses.



> It always uses the template.

Doh, yeah ok. It would probably make more sense to have this returned if `!$async` though;

```php
return static::hiddenInput($request->csrfParam, $request->getCsrfToken(), $options);
```

Reply:
If we're not going to render the input always, then yeah. I'll change.

Reply:
@brandonkelly ok js is now injecting the element and just rendering from PHP if `!$async`